test/functional/interface_rpc.py

```diff
@@ -68,17 +69,69 @@ def test_batch_request(self):
         assert_equal(result_by_id[1]['error'], None)
         assert_equal(result_by_id[1]['result'], 0)
 
-        assert_equal(result_by_id[2]['error']['code'], -32601)
+        assert_equal(result_by_id[2]['error']['code'], -32601)          # RPC_METHOD_NOT_FOUND
+        assert_equal(result_by_id[2]['error']['message'], "Method not found")
         assert_equal(result_by_id[2]['result'], None)
 
         assert_equal(result_by_id[3]['error'], None)
         assert result_by_id[3]['result'] is not None
 
-    def test_http_status_codes(self):
-        self.log.info("Testing HTTP status codes for JSON-RPC requests...")
+        assert_equal(result_by_id[4]['error']['code'], -32600)          # RPC_INVALID_REQUEST
+        assert_equal(result_by_id[4]['error']['message'], "Missing method")
+        assert_equal(result_by_id[2]['result'], None)
 
-        expect_http_status(404, -32601, self.nodes[0].invalidmethod)
-        expect_http_status(500, -8, self.nodes[0].getblockhash, 42)
+        self.log.info("Testing basic JSON-RPC 2.0 batch request...")
+        results = self.nodes[0].batch([
+            # A basic request that will work fine.
+            {"jsonrpc": "2.0", "method": "getblockcount", "id": 5},
+            # Request that will fail.  The whole batch request should still work fine.
+            {"jsonrpc": "2.0", "method": "invalidmethod", "id": 6},
+            # Another call that should succeed.
+            {"jsonrpc": "2.0", "method": "getblockhash", "id": 7, "params": [0]},
```

Comment:
Just a behaviour quirk I noticed in mutating these a little for testing: we can actually service batch requests which consist of a mixture of 1.0 and 2.0 requests. This test itself will fail, as it's asserting that `error` is not present (which if one is changed to 1.0 it is), but the server can handle it. Not that I think it's a particularly realistic or common situation...

Reply:
another good call, thats a bug. I force-pushed with a fix and added a mixed 1.1 + 2.0 in the batch test